### PR TITLE
refactor: liveness tests

### PR DESCRIPTION
**Pull request by alexanderbez**
- Remove old legacy docker-compose-based liveness setup and tests in favor of a more streamlined use of ignite CLI.
- Fix CI workflows for `Build & Test`

### CODE REVIEW

**Comment by okwme** [comment]
ah just realized this is failing lint chcker cause you removed all the `deploy-testnet.md` docs. This is realted to this issue @alexanderbez : https://github.com/cosmos/gaia/issues/1244
Do you have an opinion about the contents of the `contrib/testnets` directory Bez? Should we just nuke all of it or is any of it still used / useful in light of https://github.com/cosmos/testnets?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -70,11 +63,42 @@ jobs:
...
+        run: |
+          curl https://get.ignite.com/cli@v0.20.0! | bash
+        if: env.GIT_DIFF
+      - name: Start Local Network via Ignite CLI
+        run: |
+          make start-localnet-ci > ignite.out 2>&1 &
+        if: env.GIT_DIFF
+      - name: Test Local Network Liveness
+        run: |
+          sleep 3m
```
Comment (yaruwangway) [inline .github/workflows/test.yml:102]: why sleep ? and sleep so long ?

File: contrib/scripts/test_localnet_liveness.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+if [ -z "$4" ]; then
+  echo "Invalid argument: missing node address"
+  echo "sh test_localnet_liveness.sh <iterations> <sleep> <num-blocks> <node-address>"
+  exit 1
+fi
+
+docker_containers=($(docker ps -q -f name=umeed --format='{{.Names}}'))
+
+while [ ${CNT} -lt $ITER ]; do
```
Comment (yaruwangway) [inline contrib/scripts/test_localnet_liveness.sh:35]: is `$ITER` used for only counting a max looping ? if this way, can  we just write `while [ ${CNT} -lt $(expr ${NUMBLOCKS} + 50) ]; do` then we do not need to pass the arg `ITER=$1`

File: .github/workflows/test.yml (hunk 1 above)
Comment (alexanderbez) [reply .github/workflows/test.yml:102]: `ignite` start a blocking process where it runs a chain, so we run the process in the background and pipe both stdout and stderr to a file. `ignite` takes a bit to startup (builds app, builds proto, etc...), so the 3m is just a safe "buffer" to wait for `ignite` to actually start the chain.

File: contrib/scripts/test_localnet_liveness.sh (hunk 2 above)
Comment (alexanderbez) [reply contrib/scripts/test_localnet_liveness.sh:35]: We've had this script in the SDK for years at this point. We could probably clean it up. I'm indifferent to it as it's pretty simple.

**Review by yaruwangway — APPROVED** [review]

**Comment by yaruwangway** [comment]
> ah just realized this is failing lint chcker cause you removed all the `deploy-testnet.md` docs. This is realted to this issue @alexanderbez : #1244 Do you have an opinion about the contents of the `contrib/testnets` directory Bez? Should we just nuke all of it or is any of it still used / useful in light of https://github.com/cosmos/testnets?

I thought it was accidentally removed, so i add the `deploy-testnet.md` back to `docs/hub-tutorials`

**Pull request merged**